### ISSUE TITLE
Belfast fails to build/run on Windows

### ISSUE THREAD

**Issue by Maren0000** [ggmolly/belfast#9]
When trying to build/run Belfast on a Windows machine or trying to build it on a Linux machine using the following variables `GOOS=windows GOARCH=amd64`, the following errors appear in the console:
```
# github.com/ggmolly/belfast/connection
connection/client.go:108:20: undefined: syscall.EpollCtl
connection/client.go:108:60: undefined: syscall.EPOLL_CTL_DEL
connection/client.go:111:26: cannot use client.FD (variable of type int) as syscall.Handle value in argument to syscall.Close
connection/client.go:119:26: cannot use client.FD (variable of type int) as syscall.Handle value in argument to syscall.Write
connection/server.go:38:45: cannot use fd (variable of type int) as syscall.Handle value in argument to syscall.Getpeername
connection/server.go:70:28: cannot use syscall.Socket(syscall.AF_INET, syscall.SOCK_STREAM | syscall.O_NONBLOCK, 0) (value of type syscall.Handle) as int value in assignment
connection/server.go:73:22: cannot use server.SocketFD (variable of type int) as syscall.Handle value in argument to syscall.Close
connection/server.go:76:33: cannot use server.SocketFD (variable of type int) as syscall.Handle value in argument to syscall.SetsockoptInt
connection/server.go:80:31: cannot use server.SocketFD (variable of type int) as syscall.Handle value in argument to syscall.SetNonblock
connection/server.go:91:24: cannot use server.SocketFD (variable of type int) as syscall.Handle value in argument to syscall.Bind
connection/server.go:91:24: too many errors
```

I don't know much about Go currently, but from the research I've done, it seems that not all of the syscall functions are available on Windows. So some of the code may need to be substituted depending on the platform. Maybe the [x/sys/windows](https://pkg.go.dev/golang.org/x/sys/windows) docs can help with that.

A current workaround for Windows users is to run Belfast on WSL2. I've tested Belfast using an Ubuntu WSL2 install, and it runs fine. The web service is still accessible through a normal web browser on Windows, and the TCP server still listens on port 80. I was able to connect to Belfast with a real iOS client with this method. This is probably really inefficient in terms of resource usage but it "works" for now.

If there are any code changes that you need me to test out on my Windows machine, I would be happy to help out with that.

**Comment by ggmolly** [comment]
Hey!

Thank you very much for the detailed issue!

Indeed, `epoll` is a Linux-only syscall, so the project won't run on MacOS either.

I believe `select` could work on Windows, MacOS, Linux, and other systems. However, I thought of using a more Go-focused approach to the problem by leveraging Go's standard library or finding a dependency that can handle all systems without requiring multiple routines for each OS.

I'll be working on that very very soon!

**Comment by ggmolly** [comment]
I made some changes to fix the Windows compilation problem. I tested it on Linux, and it seems to work fine. Could you test it on Windows for me?
You can find it on the [feat/portable-tcp-server](https://github.com/ggmolly/belfast/tree/feat/portable-tcp-server) branch.

Thanks!

**Comment by Maren0000** [comment]
Seems all good here as well. Server runs on Windows, and my iOS client can connect and reach the main menu. Only minor issue I can see is an error popping up for one of the handlers (`SC_12101 - proto: required field belfast.GROUPINFO.name not set`), but it doesn't look like the game cares about that much.

Thank you for fixing this issue! This makes editing and testing the code on a Windows machine much easier. Once I learn a bit more about Go programming and AL's proto structs, I will hopefully be able to contribute to the project directly with some more handlers.

**Comment by ggmolly** [comment]
Awesome!

The proto error was caused by an incomplete implementation of a message, specifically due to an unspecified required field.

I'll test the new code further and might make some adjustments before merging the fix.
Thanks again for the report!

**Comment by ggmolly** [comment]
fixed in #10 -- [2e473de](https://github.com/ggmolly/belfast/commit/2e473de1a875b006438ad4f82603d84a254ae3d3)
